src/core/adapters/adapthighlightinfoarray.js

```diff
@@ -0,0 +1,62 @@
+import HighlightedValue from '../models/highlightedvalue';
+
+/**
+ * Constructs a highlighted data object with highlighting applied from an array of
+ * answers-core HighlightInfo models
+ *
+ * @param {HighlightInfo[]} highlightInfoArray from answers-core
+ * @returns {Object<string, string|object>} The object is keyed by fieldName, and it may consist of nested fields
+ *
+ * Example return object:
+ *
+ * {
+ *   name: '<strong>Yext</strong>',
+ *   description: {
+ *     featured: '<strong>Yext</strong> is the offical answers company'
+ *   }
+ * }
+ */
+export default function adaptHighlightInfoArray (highlightInfoArray) {
+  if (highlightInfoArray === undefined || highlightInfoArray.length === 0) {
+    return {};
+  }
+
+  return highlightInfoArray.reduce((highlightedData, highlightInfo) => {
+    const highlightedValue = new HighlightedValue()
+      .buildHighlightedValue(highlightInfo.value, highlightInfo.matchedSubstrings);
```

Comment:
are we intentionally not passing things into the constructor and directly calling buildHiglightedValue()?

Reply:
ya, the `HighlightedValue` class is a little wonky. I wasn't sure if refactoring it was outside the scope of this PR